### ISSUE TITLE
Slide number in `?print-pdf` of deck with vertical slides is wrong

### ISSUE THREAD

**Issue by michaelmyc** [hakimel/reveal.js#3198]
I'm using latest 4.3.1 release of reveal.js, and I'm trying to enable slide numbers in a deck with vertical slides. The slide numbers work fine in presentation mode, but the `?print-pdf` generated page would jump numbers when encountering vertical slides. 

I'm using the following html to create vertical slides:
```html
<section>
    Horizontal Slide 1
</section>
<section>
    <section>
        Vertical Slide 1
    </section>
    <section>
        Vertical Slide 2
    </section>
</section>
```

I'm expecting to see "1" for "Horizontal Slide 1", "2" for "Vertical Slide 1", and "3" for "Vertical Slide 2". However, I'm seeing "1", "3", and "4" respectively. The slide number also doesn't respect the custom `"c/t"` config I placed in `Reveal.initialize()`. I'm suspecting the slide numbers in `?print-pdf` is generated separately by numbering the `<section>` and rendering those that has content. This seems to be the wrong logic to me if vertical slides are present. Also, I'd love to see the pdf line numbers to respect line number formatting. 

<img width="326" alt="image" src="https://user-images.githubusercontent.com/15188392/164884130-c8f4f9cd-fe46-4908-ae9d-c109f87631cd.png">

**Comment by michaelmyc** [comment]
A quick scan of the source code confirms my suspicions. I'll take a look at how complex this fix would be. I'll open a PR if it seems simple to solve.

**Comment by hakimel** [comment]
This was recently fixed in master but isn't available in a release yet, see #3182

**Comment by kamua** [comment]
#3182 fixes the PDF page numbering for fragment group, but not for vertical slides in format "h . v".
